GCRCatalogs/user_config_mgr.py

```diff
@@ -0,0 +1,127 @@
+import os
+import warnings
+import yaml
+from collections.abc import MutableMapping
+
+"""
+Utility class for managing user config (a dict) persisted to a yaml file. 
+"""
+
+all = ['UserConfigManager']
+
+
+class UserConfigManager(MutableMapping):
+    _default_config_filename = "gcr-catalogs.yaml"
+    _default_config_reldir = "lsstdesc"
+    
+    def __init__(self, config_filename=None, config_reldir=None):
+        config_name = os.path.basename(config_filename or self._default_config_filename)
+        reldir = config_reldir or self._default_config_reldir
+        if os.path.isabs(reldir):
+            raise ValueError(f'{reldir} must be a relative path')
+        self._user_config_dir = os.path.join(self._get_config_home(), reldir)
+        self._user_config_path = os.path.join(self._user_config_dir, config_name)
+
+    @staticmethod
+    def _get_config_home():
+        if os.getenv("XDG_CONFIG_HOME"):                   # Unix
+            return(os.getenv("XDG_CONFIG_HOME"))	
+        elif os.getenv("LOCALAPPDATA"):                     # Win
+            return(os.getenv("LOCALAPPDATA"))
+
+        return os.path.join(os.path.expanduser("~"), ".config")
+
+    # Override methods which may access more than one item for efficiency
+    def keys(self):
+        return self._load_config().keys()
+
+    def values(self):
+        return self._load_config().values()
+
+    def items(self):
+        return self._load_config().items()
+
+    def update(self, items):
+        """
+        Write one or more key-value pairs to the config file.  Create new file or
+        append to existing file. 
+
+        Parameters
+        ----------
+        items       dict of entries to be written to file
+
+        Returns
+        -------
+        dict written to the file.  
+        If overwrite condition is violated, raise ValueError
+
+        """
+        config_dict = {}
+
+        if self._user_config_exists():
+            config_dict = self._load_config()
```

Comment:
`_user_config_exists` is already checked in `_load_config1
```suggestion
        config_dict = self._load_config()
```